src/components/DropZone/DropZone.tsx

```diff
@@ -224,61 +231,73 @@ export class DropZone extends React.Component<CombinedProps, State> {
       styles.DropZone,
       outline && styles.hasOutline,
       (active || dragging) && styles.isDragging,
+      measuring && styles.isMeasuring,
       error && styles.hasError,
-      size && size === 'extraLarge' && styles.sizeExtraLarge,
-      size && size === 'large' && styles.sizeLarge,
-      size && size === 'medium' && styles.sizeMedium,
-      size && size === 'small' && styles.sizeSmall,
+      height === Size.ExtraLarge && styles.sizeExtraLarge,
+      height === Size.Large && styles.sizeLarge,
+      height === Size.Medium && styles.sizeMedium,
+      height === Size.Small && styles.sizeSmall,
     );
 
+    const dragOverlayDisplayText = width === Size.ExtraLarge &&
+      height !== Size.Small && (
+        <DisplayText size="small" element="p">
+          {overlayText}
+        </DisplayText>
+      );
+
+    const dragOverlayCaption = (width === Size.Medium ||
```

Comment:
Maybe we should save them all to const so they can have specific names, since we don't add comments it maybe easier for people touching the code later to quickly see what's being done.